[2025-01-28T00:00:25.614+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2025-01-28T00:00:25.677+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: raw_api_sales_data.store_sales_data_raw scheduled__2025-01-27T00:00:00+00:00 [queued]>
[2025-01-28T00:00:25.700+0000] {taskinstance.py:2613} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: raw_api_sales_data.store_sales_data_raw scheduled__2025-01-27T00:00:00+00:00 [queued]>
[2025-01-28T00:00:25.701+0000] {taskinstance.py:2866} INFO - Starting attempt 1 of 2
[2025-01-28T00:00:25.984+0000] {taskinstance.py:2889} INFO - Executing <Task(PythonOperator): store_sales_data_raw> on 2025-01-27 00:00:00+00:00
[2025-01-28T00:00:26.005+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70: DeprecationWarning: This process (pid=1520) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-01-28T00:00:26.010+0000] {standard_task_runner.py:72} INFO - Started process 1522 to run task
[2025-01-28T00:00:26.012+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'raw_api_sales_data', 'store_sales_data_raw', 'scheduled__2025-01-27T00:00:00+00:00', '--job-id', '80', '--raw', '--subdir', 'DAGS_FOLDER/raw/dag_load_sales_data_to_db.py', '--cfg-path', '/tmp/tmpomwic8uc']
[2025-01-28T00:00:26.016+0000] {standard_task_runner.py:105} INFO - Job 80: Subtask store_sales_data_raw
[2025-01-28T00:00:26.155+0000] {task_command.py:467} INFO - Running <TaskInstance: raw_api_sales_data.store_sales_data_raw scheduled__2025-01-27T00:00:00+00:00 [running]> on host 1de8b92bf5f9
[2025-01-28T00:00:26.325+0000] {taskinstance.py:3132} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='raw_api_sales_data' AIRFLOW_CTX_TASK_ID='store_sales_data_raw' AIRFLOW_CTX_EXECUTION_DATE='2025-01-27T00:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-01-27T00:00:00+00:00'
[2025-01-28T00:00:26.328+0000] {logging_mixin.py:190} INFO - Task instance is in running state
[2025-01-28T00:00:26.329+0000] {logging_mixin.py:190} INFO -  Previous state of the Task instance: queued
[2025-01-28T00:00:26.331+0000] {logging_mixin.py:190} INFO - Current task name:store_sales_data_raw state:running start_date:2025-01-28 00:00:25.678373+00:00
[2025-01-28T00:00:26.331+0000] {logging_mixin.py:190} INFO - Dag name:raw_api_sales_data and current dag run status:running
[2025-01-28T00:00:26.332+0000] {taskinstance.py:731} INFO - ::endgroup::
[2025-01-28T00:00:26.836+0000] {database_helpers.py:20} INFO - Connected to PostgreSQL database.
[2025-01-28T00:00:26.839+0000] {raw_store_sales_data.py:34} ERROR - Failed to insert data to a table : 'method' object is not iterable
[2025-01-28T00:00:26.840+0000] {database_helpers.py:30} INFO - Closed the database connection.
[2025-01-28T00:00:26.842+0000] {raw_store_sales_data.py:38} INFO - Database Closed
[2025-01-28T00:00:26.843+0000] {taskinstance.py:3311} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/logic/raw_store_sales_data.py", line 28, in store_api_sales_data
    )for register in data.iterrows
                     ^^^^^^^^^^^^^
TypeError: 'method' object is not iterable
[2025-01-28T00:00:26.885+0000] {logging_mixin.py:190} INFO - Task instance in failure state
[2025-01-28T00:00:26.886+0000] {logging_mixin.py:190} INFO - Task start:2025-01-28 00:00:25.678373+00:00 end:2025-01-28 00:00:26.884272+00:00 duration:1.205899
[2025-01-28T00:00:26.887+0000] {logging_mixin.py:190} INFO - Task:<Task(PythonOperator): store_sales_data_raw> dag:<DAG: raw_api_sales_data> dagrun:<DagRun raw_api_sales_data @ 2025-01-27 00:00:00+00:00: scheduled__2025-01-27T00:00:00+00:00, state:running, queued_at: 2025-01-28 00:00:00.863828+00:00. externally triggered: False>
[2025-01-28T00:00:26.888+0000] {logging_mixin.py:190} INFO - Failure caused by 'method' object is not iterable
[2025-01-28T00:00:26.890+0000] {taskinstance.py:1225} INFO - Marking task as UP_FOR_RETRY. dag_id=raw_api_sales_data, task_id=store_sales_data_raw, run_id=scheduled__2025-01-27T00:00:00+00:00, execution_date=20250127T000000, start_date=20250128T000025, end_date=20250128T000026
[2025-01-28T00:00:26.940+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2025-01-28T00:00:26.942+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 80 for task store_sales_data_raw ('method' object is not iterable; 1522)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 116, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3005, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3159, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3183, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 422, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/logic/raw_store_sales_data.py", line 28, in store_api_sales_data
    )for register in data.iterrows
                     ^^^^^^^^^^^^^
TypeError: 'method' object is not iterable
[2025-01-28T00:00:26.998+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2025-01-28T00:00:27.059+0000] {taskinstance.py:3895} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-01-28T00:00:27.067+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
